resource/js/global-search.js

```diff
@@ -0,0 +1,436 @@
+      this.languageStrings = window.SKOSMOS.language_strings
+      this.vocabStrings = window.SKOSMOS.vocab_list
+    },
+    watch: {
+      selectedLanguage (newLang) {
+        if (!newLang) return
+        const url = new URL(window.location.href)
+        if (newLang === 'all') {
+          url.searchParams.set('anylang', 'on')
+        } else {
+          url.searchParams.set('clang', newLang)
+          url.searchParams.delete('anylang')
+        }
+        window.history.replaceState({}, '', url.toString())
+      }
+    },
+    methods: {
+      autoComplete () {
+        const delayMs = 300
+
+        // when new autocomplete is fired, empty the previous result
+        this.renderedResultsList = []
+
+        // cancel the timer for upcoming API call
+        clearTimeout(this._timerId)
+        this.hideAutoComplete()
+
+        // delay call, but don't execute if the search term is not at least two characters
+        if (this.searchTerm.length > 1) {
+          this._timerId = setTimeout(() => { this.search() }, delayMs)
+        }
+      },
+      search () {
+        const mySearchCounter = this.searchCounter + 1 // make sure we can identify this search later in case of several ongoing searches
+        this.searchCounter = mySearchCounter
+        let skosmosSearchUrl = window.SKOSMOS.baseHref + 'rest/v1/search?'
+        const skosmosSearchApiParams = this.formatSearchApiParams()
+        skosmosSearchUrl += skosmosSearchApiParams.toString()
+
+        fetch(skosmosSearchUrl)
+          .then(data => data.json())
+          .then(data => {
+            if (mySearchCounter === this.searchCounter) {
+              this.renderedResultsList = data.results // update results (update cache if it is implemented)
+              this.renderResults() // render after the fetch has finished
+            }
+          })
+      },
+      formatSearchUrlParams () {
+        const params = new URLSearchParams({ q: this.searchTerm })
+        if (this.selectedLanguage === 'all') {
+          params.set('anylang', 'on')
+        } else {
+          params.set('clang', this.selectedLanguage)
+        }
+        params.set('vocabs', this.formatVocabParam())
+
+        return params
+      },
+      formatSearchApiParams () {
+        const apiSearchTerm = this.searchTerm.includes('*') ? this.searchTerm : `${this.searchTerm}*`
+        const params = new URLSearchParams({ query: apiSearchTerm, unique: true })
+        params.set('lang', this.getSearchLang())
+        params.set('vocab', this.formatVocabParam())
+
+        return params
+      },
+      formatVocabParam () {
+        const vocabs = this.getSelectedVocabs
+        return vocabs.map(voc => voc.key).join(' ')
+      },
+      notationMatches (searchTerm, notation) {
+        return notation?.toLowerCase()?.includes(searchTerm.toLowerCase()) === true
+      },
+      getSearchLang () {
+        const urlParams = new URLSearchParams(window.location.search)
```

Comment:
Not sure I understand why this state information needs to be taken from the URL, when there are other places as well (e.g. component state, DOM, SKOSMOS object). But we can talk about this later